website/blog/2025/05-01-scrape-youtube/index.md

```diff
@@ -0,0 +1,442 @@
+
+If your project requires more data than the API allows, you'll need to use crawling. Let's examine the site to develop an optimal crawling strategy.
+
+To load new elements on the page, YouTube uses infinite scrolling, similar to what we discussed in the corresponding [article](https://crawlee.dev/blog/infinite-scroll-using-python) from the [Apify](https://apify.com/) team. Let's look at how this works using [DevTools](https://developer.chrome.com/docs/devtools) and the [Network](https://developer.chrome.com/docs/devtools/network/) tab.
+
+![Load Request](img/load_request.webp)
+
+If we look at the response structure, we can see that YouTube uses [JSON](https://www.json.org) to transmit data, but its structure is quite complex to navigate.
+
+![Load Response](img/load_response.webp)
+
+Therefore, we'll use [Playwright](https://playwright.dev/python/docs/intro) for crawling, which will help us avoid parsing complex JSON responses. But if you want to practice crawling complex websites, try implementing a crawler based on an HTTP client, like in this [article](https://crawlee.dev/blog/scraping-dynamic-websites-using-python).
+
+Let's analyze the selectors for getting video links using the [Elements](https://developer.chrome.com/docs/devtools/elements/) tab:
+
+![Selectors](img/load_response.webp)
+
+It looks like we're interested in `a` tags with the attribute `id="video-title-link"`!
+
+Let's look at the video page to better understand how YouTube transmits data. As expected, we see data in JSON format.
+
+![Video Response](img/video_json.webp)
+
+If you live in a country where [GDPR](https://gdpr-info.eu/) applies, you'll need to handle the following popup before you can access the data:
+
+![GDPR](img/GDPR.webp)
+
+## 3. Configuring Crawlee
+
+Configuring Crawlee for YouTube is very similar to configuring it for [TikTok](https://crawlee.dev/blog/scrape-tiktok-python). But it will have some key differences.
+
+Since pages have infinite scrolling, we need to limit the number of elements we want to get. For this, we'll add a `max_items` parameter that will limit the maximum number of elements for each search and pass it in `user_data` when forming a [Request](https://www.crawlee.dev/python/api/class/Request).
+
+We'll limit the intensity of scraping by setting `max_tasks_per_minute` in [`ConcurrencySettings`](https://www.crawlee.dev/python/api/class/ConcurrencySettings). This will help us reduce the likelihood of being blocked by YouTube.
+
+Scrolling pages can take a long time, so we should increase the time limit for processing a single request using `request_handler_timeout`.
+
+Since we won't be saving images, videos, and similar media content during crawling, we can block requests to them using [`block_requests`](https://crawlee.dev/python/api/class/BlockRequestsFunction) and [`pre_navigation_hook`](https://crawlee.dev/python/api/class/PlaywrightCrawler#pre_navigation_hook).
```

Comment:
ok know i know that we are not doing images videos or any media, but what are we doing?